tests/test_url.py

```diff
@@ -501,6 +501,21 @@ def test_with_path_fragment():
     assert str(url.with_path('/test')) == 'http://example.com/test'
 
 
+def test_with_path_empty():
+    url = URL('http://example.com/test')
+    assert str(url.with_path('')) == 'http://example.com'
+
+
+def test_with_path_slash():
+    url = URL('http://example.com/test')
+    assert str(url.with_path('/')) == 'http://example.com/'
```

Comment:
This test bothers me.
yarl shuld behave like address bar in browser.
Google Chrome autoconverts `'http://example.com/'` to `'http://example.com'`